### PR TITLE
UI fixes, adding Ron's comments, and many more things :)

### PR DESCRIPTION
**Pull request by hag007**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -148,10 +148,7 @@ app.post("/upload", timeout("10m"), (req, res, next) => {
 
         console.log(`Starting domino py execution on set ${setName}...`);
         // question -> not sure why gdocker up after cd works but not the other way around?!
-        loadBalancer=`ssh ${conf.USERNAME}@rack-shamir${serverNum}.cs.tau.ac.il cd /specific/netapp5/gaga/guests/nimsi/domino_web; gdocker up --container_name=domino; `
-        let algExecutor = loadBalancer +
-            "bash domino_runner.sh " +
-            [
+        let cmd=[
```
Comment (Nimsi123) [inline server.js:151]:
```suggestion
        let cmdArgs=[
```

File: server.js [hunk 2]
```diff
@@ -160,9 +157,17 @@ app.post("/upload", timeout("10m"), (req, res, next) => {
                 conf.DOMINO_PYTHON_ENV,
                 conf.AMI_PLUGINS_PYTHON_ENV
             ].join(" ");
-        console.log("algExecutor...\n", algExecutor);
+        localExecution=`bash domino_runner.sh ${cmd}`
```
Comment (Nimsi123) [inline server.js:160]:
```suggestion
        localExecution=`bash domino_runner.sh ${cmdArgs}`
```

File: server.js [hunk 3]
```diff
@@ -160,9 +157,17 @@ app.post("/upload", timeout("10m"), (req, res, next) => {
...
         try {
-            await execAsync(algExecutor);
+            if (conf.REMOTE_EXECUTION){            
+                console.log("About ot start remote execution")
```
Comment (Nimsi123) [inline server.js:163]:
```suggestion
                console.log("About to start remote execution")
```

File: server.js [hunk 4]
```diff
@@ -247,7 +252,7 @@ app.use((err, req, res, next) => {
   res.status(500).send('Something broke!');
 });
 
-app.listen(6500 || process.env.PORT || conf.PORT);
+app.listen(8000 || process.env.PORT || conf.PORT);
```
Comment (Nimsi123) [inline server.js:255]:
```suggestion
app.listen(conf.PORT);
```
I don't think we'll be working at the same time anymore. Also, is there a need for `process.env.PORT`?

File: src/components/LandingPage.jsx [hunk 5]
```diff
@@ -80,52 +83,57 @@ const LandingPage = ({history}) => {
...
+                return (<ResearchGroupCard cardStatus={details=="researchGroups"}/>);
             case "citation":
-                return CitationCard;
+                return (<CitationCard cardStatus={details=="citation"}/>);
             case "repositories":
-                return RepositoriesCard;
+                return (<RepositoriesCard cardStatus={details=="repositories"}/>);
             case "contact":
-                return ContactAndIssuesCard;
+                return (<ContactAndIssuesCard cardStatus={details=="contact"}/>);
```
Comment (Nimsi123) [inline src/components/LandingPage.jsx:123]: From what I can tell, we do not need the `cardStatus` prop.

The switch statement only returns a single card. In the case that `details === "repositories"`, for example, we will execute that case suite and there's no need to recompute `details === "repositories"` for cardStatus, as it is guaranteed to always be true. If `details` changes it's state, the switch statement will return a different card.

Please let me know if I'm missing anything!

File: src/components/LandingPage.jsx [hunk 6]
```diff
@@ -153,8 +161,13 @@ const LandingPage = ({history}) => {
...
 
-            <div style={{position: 'absolute', right: '110px', top: '85px', marginBottom: "100px"}}>
-                {getCard(details)}
+            <div style={{position: 'absolute', right: '110px', top: '85px', marginBottom: "100px"}} >
+                <DeveloperCreditsCard cardStatus={details=="developerCredits"}/>
+                <ResearchGroupCard cardStatus={details=="researchGroups"}/>
+                <CitationCard cardStatus={details=="citation"}/>
+                <RepositoriesCard cardStatus={details=="repositories"}/>
+                <SpecialCreditsCard cardStatus={details=="specialCredits"}/>
+                <ContactAndIssuesCard cardStatus={details=="contact"}/>
```
Comment (Nimsi123) [inline src/components/LandingPage.jsx:170]: Check out my comment above. We can revert back to `{getCard(details)}`, as this function will update the card displayed when `details` changes it's state.

File: src/components/LandingPage.jsx [hunk 7]
```diff
@@ -207,7 +219,7 @@ const LandingPage = ({history}) => {
 
                 <>
                     <h1 style={{textAlign: "center", margin: '40px 0px 40px 0px'}}>DOMINO statistics</h1>
-                    <Carousel variant="dark">
+                    <Carousel className="bg-dark">
```
Comment (Nimsi123) [inline src/components/LandingPage.jsx:222]: I couldn't get this to work! I'm glad you figured out the solution!!!

File: src/components/LandingPage.jsx [hunk 8]
```diff
@@ -80,52 +83,57 @@ const LandingPage = ({history}) => {
...
         );
     };
 
     const getCard = (cardDetail) => {
+
+
         switch (cardDetail) {
             case "developerCredits":
-                return DeveloperCredits;
+                return (<DeveloperCreditsCard cardStatus={details=="developerCredits"}/>);
```
Comment (Nimsi123) [inline src/components/LandingPage.jsx:115]:
```suggestion
                return (<DeveloperCreditsCard cardStatus={details==="developerCredits"}/>);
```
Let's try to use the `===` operator if you decide to keep this code.

**Review by Nimsi123 — CHANGES_REQUESTED** [review]

File: src/components/LandingPage.jsx (hunk 7 above)
Comment (hag007) [reply src/components/LandingPage.jsx:222]: 10x :)

File: server.js (hunk 4 above)
Comment (hag007) [reply server.js:255]:
```suggestion
app.listen(conf.PORT || 8000);
```

Comment (hag007) [reply server.js:255]: I think this is a part from a legacy code. Let me knwo if this change makes sense to you

File: src/components/LandingPage.jsx (hunk 5 above)
Comment (hag007) [reply src/components/LandingPage.jsx]: Actually this entire function should be deleted. Let's discuss this in our next meeting.

File: src/components/LandingPage.jsx (hunk 8 above)
Comment (hag007) [reply src/components/LandingPage.jsx]: True, but this entire "getCard" function should be removed altogether. Let's discuss this in our next meeting.

**Pull request merged**